src/server/service/kv.rs

```diff
@@ -1567,12 +1581,17 @@ fn future_scan<E: Engine, L: LockManager, F: KvFormat>(
 fn future_batch_get<E: Engine, L: LockManager, F: KvFormat>(
     storage: &Storage<E, L, F>,
     mut req: BatchGetRequest,
+    configs: &AtomicConfigs,
 ) -> impl Future<Output = ServerResult<BatchGetResponse>> {
-    let tracker = GLOBAL_TRACKERS.insert(Tracker::new(RequestInfo::new(
-        req.get_context(),
-        RequestType::KvBatchGet,
-        req.get_version(),
-    )));
+    let tracker = if configs.get_track_request() {
```

Comment:
Could you wrapper the code block `let tracker = if {} else {}` to a function? I found several duplicated code.

Reply:
The types are different, so it's sadly impossible to extract the whole pattern as a function. And even the function names are not totally consistent, a macro cannot solve the problems either.

Reply:
Ah, if you only want to eliminate the if/else but still extract the fields manually, it's possible.